ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/util/SqlExecutor.scala

```diff
@@ -46,23 +46,29 @@ final class SqlExecutor(noOfThread: Int, loggerFactory: NamedLoggerFactory, mm:
       waitAllTimer.update(waitNanos, TimeUnit.NANOSECONDS)
       val startExec = System.nanoTime()
       try {
-
         // Actual execution
-        val res = block
+        promise.success(block)
```

Comment:
I may be wrong here, but if `promise` is immutable then you'll have to return it from this expression and then call `.future` on that in the end.

Reply:
I believe promises are necessarily mutable; you return the `promise.future` immediately and then can complete it later.

Reply:
According to [the Scala documentation](https://docs.scala-lang.org/overviews/core/futures.html#promises), it's mutable by design.

Reply:
Indeed